server/scripts/remove-duplicate-locations.ts

```diff
@@ -106,22 +112,27 @@ function mergeGroupsWithCommonLocations(groups) {
  * @param {boolean} unpadIds If true, remove zero-padding from numeric IDs
  *        before using them to find duplicates.
  */
-function groupDuplicateLocations(locations, systems = null, unpadIds = false) {
+export function groupDuplicateLocations(
+  locations: ScriptLocation[],
+  systems: string[] = null,
+  unpadIds = false
+): Set<ScriptLocation>[] {
   const byExternalId = groupByExternalId(locations, systems, unpadIds);
   return mergeGroupsWithCommonLocations(byExternalId);
 }
 
 /**
  * Create a merge plan for each group that needs to be merged.
- * @param {Array<Set>} mergeGroups
  */
-function planChanges(mergeGroups) {
+function planChanges(mergeGroups: Set<ScriptLocation>[]) {
   // Plan merges for each group.
   const plans = [];
   for (const group of mergeGroups) {
     // Sort by oldest first, so that the oldest becomes the target all the
     // others merge into.
-    const sorted = [...group].sort((a, b) => a.created_at - b.created_at);
+    const sorted = [...group].sort(
+      (a, b) => a.created_at.getTime() - b.created_at.getTime()
+    );
```

Comment:
Adding `.getTime()` is not technically required here, but TypeScript doesn’t like subtracting `Date` instances. This is, in some very technical sense, more correct. 🤷